[CI] AssertionError: Accounting breaker not reset

CI output: https://internal-ci.elastic.co/job/elastic+x-pack-elasticsearch+6.2+multijob-windows-compatibility/682/console

```
REPRODUCE WITH: gradlew :x-pack-elasticsearch:qa:smoke-test-monitoring-with-watcher:integTestRunner \
  -Dtests.seed=9700B8DD0A76FCC8 \
  -Dtests.class=org.elasticsearch.smoketest.PreventFailingBuildIT \
  -Dtests.method="testSoThatTestsDoNotFail" \
  -Dtests.security.manager=true \
  -Dtests.locale=en-IE \
  -Dtests.timezone=Africa/Tripoli
```

Stack trace:

```
java.lang.AssertionError: Accounting breaker not reset to 0 on node: {node-0}{V8XMZi9FQLOkQk79I8dxvw}{raz_TS_pQISiiXGUBcPMuA}{127.0.0.1}{127.0.0.1:21360}{testattr=test}
Expected: <0L>
     but: was <2250L>
	at __randomizedtesting.SeedInfo.seed([9700B8DD0A76FCC8:3B7E65085E8C56E4]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.elasticsearch.test.ExternalTestCluster.ensureEstimatedStats(ExternalTestCluster.java:171)
	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:93)
	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:582)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2085)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1713)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:965)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:916)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:802)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:852)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:748)
```

I see nothing else suspicious in the test output or in the logs on the worker. Sorry, not a lot to go on here.

This happened in a different test today: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/6494/consoleFull

```
14:45:41   2> REPRODUCE WITH: ./gradlew :x-pack:plugin:rollup:internalClusterTest -Dtests.seed=9C130591FD311456 -Dtests.class=org.elasticsearch.xpack.rollup.RollupIT -Dtests.method="testTwoJobsStartStopDeleteOne" -Dtests.security.manager=true -Dtests.locale=el-CY -Dtests.timezone=Brazil/Acre
14:45:41 FAILURE 1.76s | RollupIT.testTwoJobsStartStopDeleteOne <<< FAILURES!
14:45:41    > Throwable #1: java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_sd3
14:45:41    > Expected: <0L>
14:45:41    >      but: was <4775L>
14:45:41    > 	at __randomizedtesting.SeedInfo.seed([9C130591FD311456:BA93A46A17310921]:0)
14:45:41    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
14:45:41    > 	at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2016)
14:45:41    > 	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:93)
14:45:41    > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2047)
14:45:41    > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:590)
14:45:41    > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2163)
```

And again on the rollup test:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/2014/console

```
./gradlew :x-pack:plugin:rollup:internalClusterTest \
  -Dtests.seed=1D037D5E37846F9 \
  -Dtests.class=org.elasticsearch.xpack.rollup.RollupIT \
  -Dtests.method="testTwoJobsStartStopDeleteOne" \
  -Dtests.security.manager=true \
  -Dtests.locale=es-CL \
  -Dtests.timezone=Pacific/Truk
```

The assertion is this:

```
16:24:57 FAILURE 7.00s | RollupIT.testTwoJobsStartStopDeleteOne <<< FAILURES!
16:24:57    > Throwable #1: java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_s0
16:24:57    > Expected: <0L>
16:24:57    >      but: was <7042L>
16:24:57    > 	at __randomizedtesting.SeedInfo.seed([1D037D5E37846F9:2750962E09785B8E]:0)
16:24:57    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
16:24:57    > 	at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2023)
16:24:57    > 	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:93)
16:24:57    > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2054)
16:24:57    > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:586)
16:24:57    > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2153)
16:24:57    > 	at java.lang.Thread.run(Thread.java:748)
```

same again

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=centos/2536/console

```
00:51:11 FAILURE 6.42s | RollupIT.testTwoJobsStartStopDeleteOne <<< FAILURES!
00:51:11    > Throwable #1: java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_s0
00:51:11    > Expected: <0L>
00:51:11    >      but: was <11421L>
00:51:11    > 	at __randomizedtesting.SeedInfo.seed([698DCF0963902073:4F0D6EF289903D04]:0)
00:51:11    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
00:51:11    > 	at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2020)
00:51:11    > 	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:93)
00:51:11    > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2051)
00:51:11    > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:586)
00:51:11    > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2159)
00:51:11    > 	at java.lang.Thread.run(Thread.java:748)
```

Another instance https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java10,nodes=virtual&&linux/158/console

> 16:10:08 FAILURE 4.58s J1 | ShrinkIndexIT.testShrinkIndexPrimaryTerm <<< FAILURES!
16:10:08    > Throwable #1: java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_s2
16:10:08    > Expected: <0L>
16:10:08    >      but: was <1322L>
16:10:08    > 	at __randomizedtesting.SeedInfo.seed([2A7F48E21D6F8FBF:F37D7B03379AF2]:0)
16:10:08    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
16:10:08    > 	at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2020)
16:10:08    > 	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:93)
16:10:08    > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2051)
16:10:08    > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:592)
16:10:08    > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2185)
16:10:08    > 	at jdk.internal.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
16:10:08    > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
16:10:08    > 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
16:10:08    > 	at java.base/java.lang.Thread.run(Thread.java:844)

Failed again a few times over the weekend https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=fedora/1168/console This time in different ShrinkIndexIT tests.

```
java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_sd2
Expected: <0L>
     but: was <1254L>
	at __randomizedtesting.SeedInfo.seed([814E471A446BC55F:ABC272835A33D012]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2013)
	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:93)
	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2044)
	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:592)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2179)
	at sun.reflect.GeneratedMethodAccessor21.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1713)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:965)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:916)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:802)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:852)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:863)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:748)
```

Piling another one here: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/223/consoleFull
for the `BasicWatcherTests.testConditionSearchWithIndexedTemplate` test.
Relevant log lines:
```
05:52:24   1> [2018-08-07T13:52:22,004][INFO ][o.e.x.w.t.i.BasicWatcherTests] [testConditionSearchWithIndexedTemplate] after test
05:52:24 FAILURE 1.10s J1 | BasicWatcherTests.testConditionSearchWithIndexedTemplate <<< FAILURES!
05:52:24    > Throwable #1: java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_s0
05:52:24    > Expected: <0L>
05:52:24    >      but: was <740L>
05:52:24    > 	at __randomizedtesting.SeedInfo.seed([831554F4DAAB5B36:27A252D4EB6139B8]:0)
05:52:24    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
05:52:24    > 	at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2012)
05:52:24    > 	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:93)
05:52:24    > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2043)
05:52:24    > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:593)
05:52:24    > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2186)
05:52:24    > 	at java.lang.Thread.run(Thread.java:748)
```
and this happened earlier (might be misleading, though):
```
05:52:24   1> org.elasticsearch.env.ShardLockObtainFailedException: [.watches][0]: obtaining shard lock timed out after 0ms
05:52:24   1> 	at org.elasticsearch.env.NodeEnvironment$InternalShardLock.acquire(NodeEnvironment.java:671) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.env.NodeEnvironment.shardLock(NodeEnvironment.java:590) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.env.NodeEnvironment.lockAllForIndex(NodeEnvironment.java:536) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.env.NodeEnvironment.deleteIndexDirectorySafe(NodeEnvironment.java:487) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.indices.IndicesService.deleteIndexStoreIfDeletionAllowed(IndicesService.java:722) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.indices.IndicesService.deleteIndexStore(IndicesService.java:709) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.indices.IndicesService.removeIndex(IndicesService.java:615) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.indices.cluster.IndicesClusterStateService.deleteIndices(IndicesClusterStateService.java:284) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:218) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at org.elasticsearch.cluster.service.ClusterApplierService.lambda$callClusterStateAppliers$6(ClusterApplierService.java:492) ~[elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
05:52:24   1> 	at java.lang.Iterable.forEach(Iterable.java:75) [?:1.8.0_181]
05:52:24   1> 	at org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:489) [elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/362/consoleText contains another case of this:

```
   > Throwable #1: java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_s1, are there still Lucene indices around?
   > Expected: <0L>
   >      but: was <2508L>
   > 	at __randomizedtesting.SeedInfo.seed([4AA7930E8DA3CCA6:602BA69793FBD9EB]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.test.InternalTestCluster.ensureEstimatedStats(InternalTestCluster.java:2458)
   > 	at org.elasticsearch.test.TestCluster.assertAfterTest(TestCluster.java:94)
   > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2493)
   > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:593)
   > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2191)
   > 	at java.lang.Thread.run(Thread.java:748)
```

`SplitIndexIT` failed with this error. 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+internalClusterTest/2354/console

The assertBusy did not help:
```
java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_s0, are there still Lucene indices around?
...

Expected: <0L>
     but: was <1326L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.elasticsearch.test.InternalTestCluster.lambda$ensureEstimatedStats$31(InternalTestCluster.java:2419)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 40 more
	Suppressed: java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_s0, are there still Lucene indices around?
Expected: <0L>
     but: was <1326L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.elasticsearch.test.InternalTestCluster.lambda$ensureEstimatedStats$31(InternalTestCluster.java:2419)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 40 more
```

Here's the reproduce line 

```
./gradlew :server:integTest \
  -Dtests.seed=F93D4074E6F9AF6F \
  -Dtests.class=org.elasticsearch.action.admin.indices.create.SplitIndexIT \
  -Dtests.method="testSplitIndexPrimaryTerm" \
  -Dtests.security.manager=true \
  -Dtests.locale=es-CR \
  -Dtests.timezone=America/Argentina/San_Juan \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Some more failures in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=zulu11,nodes=immutable&&linux&&docker/127/console, this time in `ShrinkIndexIT`:

```
  2> java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_sd4, are there still Lucene indices around?
    Expected: <0L>
         but: was <1394L>
```

one more in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java12,nodes=immutable&&linux&&docker/134/consoleFull for SplitIndexIT

```
org.elasticsearch.action.admin.indices.create.SplitIndexIT > testSplitIndexPrimaryTerm FAILED
    java.lang.AssertionError: Accounting breaker not reset to 0 on node: node_sd3, are there still Lucene indices around?
    Expected: <0L>
         but: was <1256L>
REPRODUCE WITH: ./gradlew :server:integTest --tests "org.elasticsearch.action.admin.indices.create.SplitIndexIT.testSplitIndexPrimaryTerm" -Dtests.seed=BCF154492FFC0AA9 -Dtests.security.manager=true -Dtests.locale=sq-MK -Dtests.timezone=America/Toronto -Dcompiler.java=12 -Druntime.java=12
```

--- a/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
+++ b/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
@@ -81,6 +81,7 @@ import org.elasticsearch.common.Randomness;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.TriFunction;
 import org.elasticsearch.common.UUIDs;
+import org.elasticsearch.common.breaker.CircuitBreaker;
 import org.elasticsearch.common.bytes.BytesArray;
 import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.collect.Tuple;
@@ -154,6 +155,7 @@ import java.util.Set;
 import java.util.concurrent.BrokenBarrierException;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.Semaphore;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
@@ -5635,5 +5637,45 @@ public class InternalEngineTests extends EngineTestCase {
         indexing.join();
         rollTranslog.join();
         assertMaxSeqNoInCommitUserData(engine);
+    }
+
+    @AwaitsFix(bugUrl = "https://issues.apache.org/jira/browse/LUCENE-8809")
+    public void testRefreshAndFailEngineConcurrently() throws Exception {
+        AtomicBoolean stopped = new AtomicBoolean();
+        Semaphore indexedDocs = new Semaphore(0);
+        Thread indexer = new Thread(() -> {
+            while (stopped.get() == false) {
+                String id = Integer.toString(randomIntBetween(1, 100));
+                try {
+                    engine.index(indexForDoc(createParsedDoc(id, null)));
+                    indexedDocs.release();
+                } catch (IOException e) {
+                    throw new AssertionError(e);
+                } catch (AlreadyClosedException e) {
+                    return;
+                }
+            }
+        });
+
+        Thread refresher = new Thread(() -> {
+            while (stopped.get() == false) {
+                try {
+                    engine.refresh("test", randomFrom(Engine.SearcherScope.values()), randomBoolean());
+                } catch (AlreadyClosedException e) {
+                    return;
+                }
+            }
+        });
+        indexer.start();
+        refresher.start();
+        indexedDocs.acquire(randomIntBetween(1, 100));
+        try {
+            engine.failEngine("test", new IOException("simulated error"));
+        } finally {
+            stopped.set(true);
+            indexer.join();
+            refresher.join();
+        }
+        assertThat(engine.config().getCircuitBreakerService().getBreaker(CircuitBreaker.ACCOUNTING).getUsed(), equalTo(0L));
     }
 }

--- a/test/framework/src/main/java/org/elasticsearch/index/engine/EngineTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/index/engine/EngineTestCase.java
@@ -55,6 +55,7 @@ import org.elasticsearch.common.CheckedBiFunction;
 import org.elasticsearch.common.Nullable;
 import org.elasticsearch.common.Randomness;
 import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.breaker.CircuitBreaker;
 import org.elasticsearch.common.bytes.BytesArray;
 import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.compress.CompressedXContent;
@@ -269,6 +270,8 @@ public abstract class EngineTestCase extends ESTestCase {
                 assertConsistentHistoryBetweenTranslogAndLuceneIndex(replicaEngine, createMapperService("test"));
                 assertMaxSeqNoInCommitUserData(replicaEngine);
             }
+            assertThat(engine.config().getCircuitBreakerService().getBreaker(CircuitBreaker.ACCOUNTING).getUsed(), equalTo(0L));
+            assertThat(replicaEngine.config().getCircuitBreakerService().getBreaker(CircuitBreaker.ACCOUNTING).getUsed(), equalTo(0L));
         } finally {
             IOUtils.close(replicaEngine, storeReplica, engine, store, () -> terminate(threadPool));
         }

--- a/test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java
@@ -2334,15 +2334,18 @@ public final class InternalTestCluster extends TestCluster {
                 final CircuitBreakerService breakerService = getInstanceFromNode(CircuitBreakerService.class, nodeAndClient.node);
                 CircuitBreaker fdBreaker = breakerService.getBreaker(CircuitBreaker.FIELDDATA);
                 assertThat("Fielddata breaker not reset to 0 on node: " + name, fdBreaker.getUsed(), equalTo(0L));
-                try {
-                    assertBusy(() -> {
-                        CircuitBreaker acctBreaker = breakerService.getBreaker(CircuitBreaker.ACCOUNTING);
-                        assertThat("Accounting breaker not reset to 0 on node: " + name + ", are there still Lucene indices around?",
-                            acctBreaker.getUsed(), equalTo(0L));
-                    });
-                } catch (Exception e) {
-                    throw new AssertionError("Exception during check for accounting breaker reset to 0", e);
-                }
+
+                // Mute this assertion until we have a new Lucene snapshot with https://issues.apache.org/jira/browse/LUCENE-8809.
+                // try {
+                //    assertBusy(() -> {
+                //        CircuitBreaker acctBreaker = breakerService.getBreaker(CircuitBreaker.ACCOUNTING);
+                //        assertThat("Accounting breaker not reset to 0 on node: " + name + ", are there still Lucene indices around?",
+                //            acctBreaker.getUsed(), equalTo(0L));
+                //    });
+                // } catch (Exception e) {
+                //    throw new AssertionError("Exception during check for accounting breaker reset to 0", e);
+                // }
+
                 // Anything that uses transport or HTTP can increase the
                 // request breaker (because they use bigarrays), because of
                 // that the breaker can sometimes be incremented from ping

